### PR TITLE
Update dependencies & support scala 2.13

### PR DESCRIPTION
**Pull request by cchantep**
- Update dependencies and Scala version (support 2.13)
- Fix warnings/code smells
- Update CI

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -1,25 +1,88 @@
 import Dependencies._
 
+inThisBuild(List(
```
Comment (cchantep) [inline build.sbt:3]: Update build to support Scala 2.13 and with property compilation options

File: project/Dependencies.scala [hunk 2]
```diff
@@ -1,17 +1,17 @@
 import sbt._
 
 object Dependencies {
+  val scalacheckVersion = "1.14.3"
```
Comment (cchantep) [inline project/Dependencies.scala:4]: Updated dependencies

File: project/Scapegoat.scala [hunk 3]
```diff
@@ -0,0 +1,54 @@
+import sbt.Keys._
```
Comment (cchantep) [inline project/Scapegoat.scala:1]: Scapegoat configuration to check code smell at compilation

File: src/main/scala/com/stackstate/pac4j/AkkaHttpSecurity.scala [hunk 4]
```diff
@@ -76,24 +76,28 @@ class AkkaHttpSecurity(config: Config, sessionStorage: SessionStorage, val sessi
 
   import AkkaHttpSecurity._
 
+  @SuppressWarnings(Array("AsInstanceOf"))
```
Comment (cchantep) [inline src/main/scala/com/stackstate/pac4j/AkkaHttpSecurity.scala:79]: For now

File: src/main/scala/com/stackstate/pac4j/AkkaHttpWebContext.scala [hunk 5]
```diff
@@ -67,7 +67,7 @@ case class AkkaHttpWebContext(request: HttpRequest,
       name = cookie.getName,
       value = cookie.getValue,
       expires = None,
-      maxAge = if (cookie.getMaxAge < 0) None else Some(cookie.getMaxAge),
+      maxAge = if (cookie.getMaxAge < 0) None else Some(cookie.getMaxAge.toLong),
```
Comment (cchantep) [inline src/main/scala/com/stackstate/pac4j/AkkaHttpWebContext.scala:70]: Fix numeric widening

File: src/main/scala/com/stackstate/pac4j/AkkaHttpWebContext.scala [hunk 6]
```diff
@@ -97,9 +97,9 @@ case class AkkaHttpWebContext(request: HttpRequest,
     changes = changes.copy(headers = header :: changes.headers.filter(_.name != name))
   }
 
-  override def getRequestParameters: java.util.Map[String, Array[String]] = {
-    requestParameters.mapValues(Array(_)).asJava
-  }
+  @com.github.ghik.silencer.silent("mapValues")
```
Comment (cchantep) [inline src/main/scala/com/stackstate/pac4j/AkkaHttpWebContext.scala:100]: Scala 2.13 collection API changes

File: src/main/scala/com/stackstate/pac4j/AkkaHttpWebContext.scala [hunk 7]
```diff
@@ -171,7 +171,10 @@ case class AkkaHttpWebContext(request: HttpRequest,
     addResponseCookie(cookie)
   }
 
-  def addResponseCsrfCookie(): Unit = CsrfCookieAuthorizer(this, Some(sessionStorage.sessionLifetime))
+  def addResponseCsrfCookie(): Unit = {
+    CsrfCookieAuthorizer(this, Some(sessionStorage.sessionLifetime))
```
Comment (cchantep) [inline src/main/scala/com/stackstate/pac4j/AkkaHttpWebContext.scala:175]: Discarded unit-value

File: src/main/scala/com/stackstate/pac4j/AuthenticatedRequest.scala [hunk 8]
```diff
@@ -6,12 +6,16 @@ import org.pac4j.core.profile.UserProfile
...
+case class AuthenticatedRequest private[pac4j] (
+  private[pac4j] val webContext: AkkaHttpWebContext,
+  /*
+   * Profiles can be accessed such that the user can inspect data 
+   * from the authentication. Should not be empty after authentication
+   */
+  val profiles: List[UserProfile]) {
 
-  lazy val mainProfile: UserProfile = profiles.head
+  lazy val mainProfile: UserProfile = profiles.headOption match {
```
Comment (cchantep) [inline src/main/scala/com/stackstate/pac4j/AuthenticatedRequest.scala:17]: Avoid `.head` with explicit exception. Would deprecate this `mainProfile` and have a `Option[UserProfile]` instead

File: src/test/scala/com/stackstate/pac4j/AkkaHttpSecurityTest.scala [hunk 9]
```diff
@@ -150,7 +156,7 @@ class AkkaHttpSecurityTest extends WordSpecLike with Matchers with ScalatestRout
       } ~> check {
         status shouldEqual StatusCodes.OK
         responseAs[String] shouldBe "called!"
-        headers.size shouldBe 1
+        headers.size shouldBe 2
```
Comment (hierynomus) [inline src/test/scala/com/stackstate/pac4j/AkkaHttpSecurityTest.scala:159]: Why are there 2 now?

File: project/Dependencies.scala [hunk 10]
```diff
@@ -1,17 +1,17 @@
...
+  val akkaHttpVersion = "10.2.0"
+  val akkaStreamsVersion = "2.6.9"
```
Comment (hierynomus) [inline project/Dependencies.scala:6]: I'm slightly hesitant to upgrade the Akka dependencies without testing them against the product in which we use this library, let's separate the concerns of doing cross compile and upgrading dependencies in 2 PRs

**Review by hierynomus — CHANGES_REQUESTED** [review]

File: src/test/scala/com/stackstate/pac4j/AkkaHttpSecurityTest.scala (hunk 9 above)
Comment (cchantep) [reply src/test/scala/com/stackstate/pac4j/AkkaHttpSecurityTest.scala:159]: Akka-HTTP no longer prevent duplicate from what I see

File: project/Dependencies.scala (hunk 10 above)
Comment (cchantep) [reply project/Dependencies.scala:6]: Generally speaking, Akka libs are compatible from 2.4 to 2.6, but I can separate that.

**Review by hierynomus — APPROVED** [review]

**Comment by hierynomus** [comment]
Just noticed one more thing, we're using GH Actions nowadays, not the Travis build anymore. Can you update the flow to do the crosscompile and scapegoat also?

Thanks!

**Comment by cchantep** [comment]
Scapegoat doesn't support 2.13

**Comment by cchantep** [comment]
done

**Comment by cchantep** [comment]
Up

**Comment by cchantep** [comment]
up

**Comment by cchantep** [comment]
Up

**Pull request merged**